ref(native): Clarify before_send with Crashpad (#5092)
getsentry/sentry-docs: src/includes/configuration/before-send/native.mdx

The callback is executed in the same thread as the call to `sentry_capture_event`. Work performed by the function may thus block the executing thread. For this reason, consider avoiding heavy work in `before_send`.

<Alert level="warning" title="Not Supported in Crashpad on macOS">

> The Crashpad backend on macOS currently has no support for notifying the crashing process and can thus not correctly terminate sessions or call the registered `before_send` hook. It will also lose any events queued for sending at the time of the crash.

Reviewer: ```suggestion
The Crashpad backend on macOS doesn't currently support notifying the crashing process and thus can't correctly terminate sessions or call the registered `before_send` hook. It will also lose any events queued for sending at the time of the crash.
```
